src/lib/pickles/impls.ml

```diff
@@ -144,6 +144,24 @@ module Step = struct
     in
     let typ = Typ.transport typ ~there:to_data ~back:of_data in
     Spec.ETyp.T (typ, (fun x -> of_data (f x)), fun x -> f_inv (to_data x))
+
+  let lock = ref (Promise.return ())
+
+  let run_async_circuit (f : unit -> 'a Promise.t) : 'a Promise.t =
```

Comment:
Don't do this. We need to be able to run a circuit inside an `exists` block, which this breaks.

Reply:
Reverted and fixed tests differently 👍🏻